lib/profile/commands/configure.rb

```diff
@@ -31,18 +36,29 @@ def display_details
       def ask_questions
         raise "No valid cluster types available" if !Type.all.any?
         type = cluster_type
-        @answers = prompt.collect do
-          type.questions.each do |question|
-            key(question.id).ask(question.text) do |q|
-              if type.fetch_answer(question.id)
-                q.default type.fetch_answer(question.id)
-              elsif question.default
-                q.default question.default
-              end
-              q.required question.validation.required
-              if question.validation.to_h.key?(:format)
-                q.validate Regexp.new(question.validation.format)
-                q.messages[:valid?] = question.validation.message
+        raise "Valid cluster type not provided" if !type
+        if @options.answers
+          given = @answers.keys
+          required = @type.questions.each.map(&:id)
+          if !(required - given).empty?
+            raise "The following questions were not answered by the JSON data: #{(required - given).join(", ")}"
+          elsif !(given - required).empty?
+            raise "The following given answers are not recognised by the cluster type: #{(given - required).join(", ")}"
+          end
```

Comment:
Is `@answers` actually being set anywhere here?

Reply:
@Womblue Has this been addressed?

Reply:
Yes, previously it was just set in `run` (you can see the assignment in the resolved thread about JSON error handling)